docs/developer-notes.md

```diff
@@ -139,6 +139,14 @@ CATcher uses the OAuth 2.0 protocol to authenticate users. Below is a summary of
 
 The authentication process is kicked off in the `AuthComponent`, but the code that co-ordinates steps 1 and 2 can be found in [`oauth.ts`](../oauth.ts)(For Electron) or `AuthService`(For Web). Step 2 requires a client secret granted to CATcher. To protect this, we run a web service, [gatekeeper](https://github.com/CATcher-org/gatekeeper) that executes step 2 on behalf of the client CATcher app.
 
+## Logging
+
+CATcher caches log data in `localStorage` within the browser. `localStorage` stores data as key-value pairs and behave similarly to cookies. The caching processes is handled the following way.
```

Comment:
I think the writeup can be better organised, as there are two distinct points being made (caching and use of the custom ErrorHandlingService).

I suggest this:


CATcher caches log data in `localStorage` within the browser. The app's `LoggingService` exposes methods for logging error, warning or informational messages, and uses the `localStorage` API  under-the-hood. `LoggingService` can be used in any component / service, as needed for the app's logging purposes. `LoggingService` stores the logs for each session together, in sequence.

To ensure that all runtime errors are logged, CATcher's `ErrorHandlingService` implements the [ErrorHandler class](https://angular.io/api/core/ErrorHandler) and then invokes `LoggingService` in its `handleError` method.

What do you think @ptvrajsk ?

I removed this sentence:
`localStorage` stores data as key-value pairs and behave similarly to cookies. (devs can easily read up on `localStorage` themselves.

Reply:
Yes I think that's better organised 👍 Thanks!